src/go/rpk/pkg/api/admin/admin.go

```diff
@@ -25,15 +25,22 @@ import (
 	"sync"
 	"time"
 
+	"github.com/hashicorp/go-multierror"
+	log "github.com/sirupsen/logrus"
 	"github.com/spf13/afero"
 	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
 	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/net"
 )
 
+// ErrNoAdminAPILeader happen when there's no leader for the Admin API
+var ErrNoAdminAPILeader = fmt.Errorf("no Admin API leader found")
+
 // AdminAPI is a client to interact with Redpanda's admin server.
 type AdminAPI struct {
-	urls   []string
-	client *http.Client
+	urls           []string
+	brokerIdToUrls sync.Map // map(brokerID: int) => URL: string
```

Comment:
It may be easier to use a `map[int]string` and a mutex, in this case -- there is very little concurrent in this code path. AFAICT the mapping happens once, basically.

Reply:
🤔 the mapping happens once, but it will spawn one go routine for each broker. Since each go routine will try to write to the map concurrently I thought about using `sync.Map`. We could also use map + mutex, but why not `sync.Map` which is made for this?

Reply:
This prompted me to go educate myself about this... I think `sync.Map` has higher overhead for simple use cases that don't have much lock contention (docs say "Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.")

Reply:
fair enough. I'll do the change to regular map :)